### ISSUE TITLE
Uniformize Flash API to TorchVision / TorchVideo

### ISSUE THREAD

**Issue by tchaton** [PyTorchLightning/lightning-flash#218]
There's several formats of data we support:

1. loading encoded video paths from csv with format: "path_name, label" (PySlowFast repo uses this)
2. loading encoded video paths from directory structure as in comment (TorchVision uses this)
3. loading frame videos (i.e. a folder of images) from directory structure
4. loading frame videos from csv with a format specific to the dataset (e.g. EpicKitchen and DomSev datasets use this)

_Originally posted by @tullie in https://github.com/PyTorchLightning/lightning-flash/pull/216#discussion_r614355300_

**Comment by tchaton** [comment]
```
DataModule
    from_paths(
        cls,
        train_data_path,
        val_data_path,
        test_data_path,
        predict_data_path,
        extensions: Optional[List[str]] = None
        data_col: str = None,
        csv_parser=LabeledDataPaths
    )
        return LabeledDataPaths([(path, label)])
class LabeledDataPaths:
    def label_row(self, row):
        return map_to_label(row[data_col])
    def data_row(self, row):
        return row[data_col]
    @property
    def num_classes(self):
        ...
    cls.from_paths("./train.csv", data_col="path", label_cols=["num_feat", ",,.."])
    path  sex, ...
    path_1 0, 2, 4
    path label
    path_1, [0, 1, 0, 0]
    path, label
    path_1, 0
```
